src/main/ruby/truffleruby/core/proc.rb

```diff
@@ -128,4 +128,12 @@ def <<(other)
       end
     end
   end
+
+  def parameters(lambda: undefined)
+    unless Primitive.undefined?(lambda)
+      lambda = Primitive.as_boolean(lambda)
+    end
+
+    Primitive.proc_parameters(self, lambda)
+  end
```

Comment:
It seems `lambda: nil` will be treated like `lambda: false` but actually `lambda: nil` works like `parameters()` called without any argument:

```ruby
lambda {|a|}.parameters(lambda: false) #=> [[:opt, :a]]
lambda {|a|}.parameters(lambda: nil) #=> [[:req, :a]]
lambda {|a|}.parameters #=> [[:req, :a]]

Reply:
Great, we've updated that now and added tests for the `lambda: nil` cases.